### PR TITLE
Fix an edge case when field `missing_value` is not `None`

### PR DESCRIPTION
**Pull request by agroszer**
but a custom value that works as `None`.

### CODE REVIEW

File: src/z3c/form/tests/test_bugfix.py [hunk 1]
```diff
@@ -62,5 +62,90 @@ class TestForm(z3c.form.form.BaseForm):
...
+        return str(value)
+
+class MockLocale(object):
+    def getFormatter(self, category):
+        return MockFormatter()
+
+
+class OurNone(object):
+    # this quacks like None, but is NOT None
+    def __cmp__(self, other):
```
Comment (icemac) [inline src/z3c/form/tests/test_bugfix.py]: This is not used in Python 3.
Even in Python 2 it should be unused because there is also `__eq__`.

File: src/z3c/form/tests/test_bugfix.py [hunk 2]
```diff
@@ -62,5 +62,90 @@ class TestForm(z3c.form.form.BaseForm):
...
+        field.missing_value = OUR_NONE
+        self.assertEqual(ndc.toWidgetValue(u''), u'')
+        self.assertEqual(ndc.toWidgetValue(None), u'')
+        self.assertEqual(ndc.toWidgetValue(OUR_NONE), u'')
+        self.assertEqual(ndc.toWidgetValue([]), u'[]')
+
+
 def test_suite():
-    return unittest.makeSuite(TestApplyChangesDictDatamanager)
+    return unittest.TestSuite((
```
Comment (icemac) [inline src/z3c/form/tests/test_bugfix.py]: I'd prefer: `return unittest.defaultTestLoader.loadTestsFromName(__name__)`

File: src/z3c/form/tests/test_bugfix.py (hunk 1 above)
Comment (icemac) [reply src/z3c/form/tests/test_bugfix.py]: See https://coveralls.io/builds/21930709/source?filename=src/z3c/form/tests/test_bugfix.py#L81 for not used methods of this class.
